### ISSUE TITLE
Foreign Key violation message crashing app

### ISSUE THREAD

**Issue by GuilhermeReda** [sequelize/sequelize#2753]
```
C:\Users\Guilherme\Documents\node\iapo1\node_modules\sequelize\lib\dialects\postgres\query.js:252
    index = err.message.match(/violates foreign key constraint \"(.+?)\"/)[1];
                                                                          ^
TypeError: Cannot read property '1' of null
    at module.exports.Query.formatError (C:\Users\Guilherme\Documents\node\iapo1\node_modules\sequelize\lib\dialects\postgres\query.js:252:79)
    at null.<anonymous> (C:\Users\Guilherme\Documents\node\iapo1\node_modules\sequelize\lib\dialects\postgres\query.js:64:21)
    at emit (events.js:95:17)
    at Query.handleError (C:\Users\Guilherme\Documents\node\iapo1\node_modules\pg\lib\query.js:99:8)
    at null.<anonymous> (C:\Users\Guilherme\Documents\node\iapo1\node_modules\pg\lib\client.js:163:26)
    at emit (events.js:95:17)
    at Socket.<anonymous> (C:\Users\Guilherme\Documents\node\iapo1\node_modules\pg\lib\connection.js:109:12)
    at Socket.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:764:14)
    at Socket.emit (events.js:92:17)
18 Dec 17:15:53 - [nodemon] app crashed - waiting for file changes before starting...
```

**Comment by janmeier** [comment]
Hmm, appears to be a bug then. Sorry for the long response time @GuilhermeReda but could you post the offending code + model definition? 

And which pg version are you experiencing this on?

**Comment by FernandoCagale** [comment]
Hello,

This problem was solved? I am also going through this situation!

    "pg": "^4.3.0",
    "pg-hstore": "^2.3.2",
    "sequelize": "^2.0.5",

**Comment by BridgeAR** [comment]
Fixed by #3567. Please use version 2.1
